docs/ReferenceGuides/UnderscoredAttributes.md

```diff
@@ -0,0 +1,428 @@
+
+## `@_effects(effectname)`
+
+Tells the compiler that the implementation of the defined function is limited
+to certain side effects. The attribute argument specifies the kind of side
+effect limitations that apply to the function including any other functions
+it calls. This is used to provide information to the optimizer that it can't
+already infer from static analysis.
+
+Changing the implementation in a way that violates the optimizer's assumptions
+about the effects results in undefined behavior.
+
+For more details, see [OptimizerEffects.rst](/docs/Proposals/OptimizerEffects.rst).
+
+## `@_exported`
+
+Re-exports all declarations from an imported module.
+
+This attribute is most commonly used by overlays.
+
+```swift
+// module M
+func f() {}
+
+// module N
+@_exported import M
+
+// module P
+import N
+func g() {
+  N.f() // OK
+}
+```
+
+## `@_fixed_layout`
+
+## `@_hasInitialValue`
+
+## `@_hasMissingDesignatedInitializers`
+
+Indicates that there may be designated initializers that are
+not printed in the swiftinterface file for a particular class.
+
+This attribute is needed for the initializer model to maintain correctness when
+library evolution is enabled. This is because a class may have non-public
+designated initializers, and Swift allows the inheritance of convenience
+initializers if and only if the subclass overrides (or has synthesized
+overrides) of every designated initializer in its superclass. Consider the
+following code:
+
+```swift
+// Lib.swift
+open class A {
+  init(invisible: ()) {}
+
+  public init(visible: ()) {}
+  public convenience init(hi: ()) { self.init(invisible: ()) }
+}
+
+// Client.swift
+class B : A {
+  var x: String
+
+  public override init(visible: ()) {
+    self.x = "Garbage"
+    super.init(visible: ())
+  }
+}
+```
+
+In this case, if `B` were allowed to inherit the convenience initializer
+`A.init(invisible:)` then an instance created via `B(hi: ())` would fail
+to initialize `B.x` resulting in a memory safety hole. What's worse is
+there is no way to close this safety hole because the user cannot override
+the invisible designated initializer because they lack sufficient visibility.
+
+## `@_hasStorage`
+
+## `@_implementationOnly`
+
+Used to mark an imported module as an implementation detail.
+This prevents types from that module being exposed in API
+(types of public functions, constraints in public extension etc.)
+and ABI (usage in `@inlinable` code).
+
+## `@_implements(ProtocolName, Requirement)`
+
+An attribute that indicates that a function with one name satisfies
+a protocol requirement with a different name. This is especially useful
+when two protocols declare a requirement with the same name, but the
+conforming type wishes to offer two separate implementations.
+
+```swift
+protocol P { func foo() }
+
+protocol Q { func foo() }
+
+struct S : P, Q {
+  @_implements(P, foo())
+  func foo_p() {}
+  @_implements(Q, foo())
+  func foo_q() {}
+}
+```
```

Comment:
It may not have come across, but I gave background on `@_disfavoredOverload` to try to illustrate when it should be used (i.e. to work around a known defect in overload resolution that can't be immediately fixed without breaking source, and particularly to work around that specific bug with literals).